content/docs/command-reference/repro.md

```diff
@@ -91,6 +91,27 @@ same time (e.g. in separate terminals). After both finish successfully, you can
 then run `dvc repro train`: DVC will know that both branches are already
 up-to-date and only execute the final stage.
 
+## Arguments
+
+- `targets` - Stage name(s) to reproduce (`./dvc.yaml` by default)
+
+  Target can be a name of the stage in the `dvc.yaml` file or a path to a `.dvc`
+  or a `dvc.yaml` file. In case of a file, it will reproduce all of the stages
+  present in that file.
```

Comment:
> or a path to a `.dvc` (file)
> In case of a file, it will reproduce all of the stages present in that file.

what about .dvc files? They don't contain stages. What's reproduced in that case? I guess it depends on the contents (e.g. import .dvc files are always frozen last time I checked). I will double check all this this but any notes are appreciates @skshetry.

Reply:
> What about .dvc files?

We still support `.dvc` files as a stage (and, internally every `.dvc` file including `add/import`-ed ones are a stage). So, this must have been generalized. 

> They don't contain stages

Each `.dvc` file is a stage on itself, so it contains one stage.

> What's reproduced in that case?

The non-frozen imports are re-imported/updated. By default, they are frozen, but you can remove the `frozen` keyword.
For the `dvc add`-ed files, they are just checked out.
For the old `.dvc` files with commands, they are run. If we have a run cache, they are checked out without running.
Same in the case of stages in `dvc.yaml` files.

Reply:
I am not sure what information should be or not be in the docs. But, that's how the `dvc repro` works.

Reply:
> internally every .dvc file including add/import-ed ones are a stage
> Each .dvc file is a stage on itself, so it contains one stage.

This was kind of true in DVC 0.x but in actuality .dvc files are not stages right? Cc @shcheklein 

> p.s. whether they're called `stage` internally somewhere in the code is probably not an indication that that's what they are.

Reply:
> non-frozen imports are re-imported/updated. By default, they are frozen, but you can remove the frozen

Right I forgot you could do that, thanks for the info.

> dvc add-ed files, they are just checked out

Hmmm I think that if they have changes they are re-added by repro.

Reply:
> Each .dvc file is a stage on itself, so it contains one stage.

my take on this - this is an implementation detail. We should be going away from this. Even imports are not stages to my mind.

I would argue that if there are other commands to "reproduce" .dvc files we should deprecate and do not complicate repro with them.

Reply:
Agree but that's a disccusion for core. For now I'll keep .dvc files and stages separate comceptually, and try to include only the minimal info about using repro for .dvc files.

Reply:
@jorgeorpinel my point is that we can start deprecating it. It means we should be paying too much attention or include into docs them. Unless I'm missing something and we indeed have a case for `dvc repro .dvc`

Reply:
> dvc add-ed files, they are just checked out

@skshetry I checked this and it doesn't seem that's the case. It gives me this error:

```
ERROR: failed to reproduce 'data.dvc': missing data 'source': data
```
> The file contents are cached.

The docs now read "`dvc repro` does not run `dvc fetch`, `dvc pull` or `dvc checkout`" which seems accurate so I'll leave that as is.